extension/src/test/TaskRunner.test.ts

```diff
@@ -0,0 +1,126 @@
+import * as assert from "assert";
+import * as fs from "fs";
+import * as path from "path";
+import { TaskRunner } from "../Template/TaskRunner";
+import { TaskRunnerItem } from "../Template/TaskRunnerItem";
+
+suite("Task Runner Tests", function () {
+  const testResourcesPath = path.join(__dirname, "../../src/test/resources");
+  const tempPath = path.join(testResourcesPath, "temp");
+
+  test("TaskRunner.exportTasksRunnerItems()", function () {
+    const taskList: TaskRunnerItem[] = [
+      {
+        description: "Show release notes.",
+        command: "update.showCurrentReleaseNotes",
+      },
+      {
+        description: "Show release notes AGAIN.",
+        command: "update.showCurrentReleaseNotes",
+      },
+    ];
+    TaskRunner.exportTasksRunnerItems(taskList, tempPath);
+    assert.ok(fs.existsSync(path.join(tempPath, "001.nab.taskrunner.json")));
+    assert.ok(fs.existsSync(path.join(tempPath, "002.nab.taskrunner.json")));
+  });
+
+  test("TaskRunner.importTaskRunnerItems", function () {
+    /**
+     * Depends on TaskRunner.exportTasksRunnerItems().
+     */
+    const taskRunner = TaskRunner.importTaskRunnerItems(testResourcesPath);
+    assert.ok(taskRunner instanceof TaskRunner);
+    assert.strictEqual(taskRunner.taskList.length, 2);
+  });
+
+  test("TaskRunner.deleteTaskFile()", async function () {
+    /**
+     * Depends on TaskRunner.exportTasksRunnerItems().
+     */
+    const taskRunner = TaskRunner.importTaskRunnerItems(tempPath);
+    assert.strictEqual(
+      taskRunner.taskList.length,
+      2,
+      "Unexpected number of files."
+    );
+    taskRunner.taskList[0].command = "update.showCurrentReleaseNotes";
+    taskRunner.taskList[1].command = "update.showCurrentReleaseNotes";
+    for (const task of taskRunner.taskList) {
+      assert.ok(task.taskPath, "Task is missing task path.");
+
+      await assert.doesNotReject(async () => {
+        await taskRunner.execute(task);
+      }, "Unexpected rejection of promise.");
+
+      assert.ok(
+        !fs.existsSync(task.taskPath),
+        `Expected file to be deleted: ${task.taskPath}`
+      );
+    }
+  });
+
+  test("TaskRunner.executeTaskList", async function () {
+    const taskList: TaskRunnerItem[] = [
+      {
+        description: "Show release notes.",
+        command: "update.showCurrentReleaseNotes",
+      },
```

Comment:
Shouldn't an iteration test always have at least two items to test the iteration parts?

Reply:
done